yt/frontends/ramses/data_structures.py

```diff
@@ -460,12 +469,25 @@ def __init__(self, filename, dataset_type='ramses',
         cosmological: If set to None, automatically detect cosmological simulation. If a boolean, force
                       its value.
         '''
+
         self._fields_in_file = fields
         # By default, extra fields have not triggered a warning
         self._warned_extra_fields = defaultdict(lambda: False)
         self._extra_particle_fields = extra_particle_fields
         self.force_cosmological = cosmological
         self._bbox = bbox
+
+        # Infer if the output is organized in groups
+        root_folder, group_folder = os.path.split(os.path.split(filename)[0])
+
+        if group_folder == 'group_00001':
+            # Count the number of groups
+            self.num_groups = len(glob(os.path.join(root_folder, 'group_?????')))
```

Comment:
Only suggestion here would be to also add a check that makes sure the `group_?????` entries are directories, like:

```
groups = [_ for _ in glob(os.path.join(root_folder, 'group_?????')) if os.path.isdir(_)]
```

But I think the risk of hitting that possible error is very low.